### PR TITLE
Fix softmax selection for EnvironmentPipeline

### PR DESCRIPTION
**Pull request by sharath**
Addresses #317

### CODE REVIEW

File: examples/breakout/breakout.py [hunk 1]
```diff
@@ -49,7 +49,6 @@
     pipeline.reset_()
     is_done = False
     while not is_done:
-        # Broken until select_softmax is implemented.
```
Comment (djsaunde) [inline examples/breakout/breakout.py]: Should this be implemented now?

File: examples/breakout/breakout_stdp.py [hunk 2]
```diff
@@ -59,7 +59,6 @@ def run_pipeline(pipeline, episode_count):
         pipeline.reset_()
         is_done = False
         while not is_done:
-            # Broken until select_softmax is implemented.
```
Comment (djsaunde) [inline examples/breakout/breakout_stdp.py]: Same question here.

File: bindsnet/pipeline/environment_pipeline.py [hunk 3]
```diff
@@ -147,12 +157,17 @@ def step_(
         obs, reward, done, info = gym_batch
 
         # Place the observations into the inputs.
-        inpts = {k: obs for k in self.inpts}
+        inpts = {
+            k: obs.repeat(self.time, *([1] * len(obs.shape[1:]))) for k in self.inpts
```
[the remark this answers is not in the document]
Comment (sharath) [reply bindsnet/pipeline/environment_pipeline.py]: good catch

File: bindsnet/pipeline/environment_pipeline.py [hunk 4]
```diff
@@ -76,6 +77,15 @@ def __init__(
         self.first = True
         self.analyzer = MatplotlibAnalyzer(**self.plot_config)
 
+        if self.output is not None:
+            self.network.add_monitor(
```
[the remark this answers is not in the document]
Comment (sharath) [reply bindsnet/pipeline/environment_pipeline.py]: This _should_ be the responsibility of the `EnvironmentPipeline` class

File: bindsnet/pipeline/environment_pipeline.py [hunk 5]
```diff
@@ -147,12 +157,17 @@ def step_(
         obs, reward, done, info = gym_batch
 
         # Place the observations into the inputs.
-        inpts = {k: obs for k in self.inpts}
+        inpts = {
```
[the remark this answers is not in the document]
Comment (sharath) [reply bindsnet/pipeline/environment_pipeline.py]: Not really, it's consistent because its `inpts` everywhere

File: bindsnet/pipeline/action.py [hunk 6]
```diff
@@ -72,16 +72,9 @@ def select_softmax(pipeline: EnvironmentPipeline, **kwargs) -> int:
         pipeline, "spike_record"
```
[the remark this answers is not in the document]
Comment (sharath) [reply bindsnet/pipeline/action.py]: Done

File: bindsnet/pipeline/action.py [hunk 7]
```diff
@@ -72,16 +72,9 @@ def select_softmax(pipeline: EnvironmentPipeline, **kwargs) -> int:
...
-    _sum = torch.sum(torch.exp(spikes.float()))
-
-    if _sum == 0:
-        action = np.random.choice(pipeline.env.action_space.n)
-    else:
-        action = torch.multinomial((torch.exp(spikes.float()) / _sum).view(-1), 1)[0]
-
-    return action
+    spikes = torch.sum(pipeline.spike_record[output], dim=0)
+    probs = torch.softmax(spikes, dim=0)
```
[the remark this answers is not in the document]
Comment (sharath) [reply bindsnet/pipeline/action.py]: Done

File: examples/breakout/breakout.py (hunk 1 above)
Comment (sharath) [reply examples/breakout/breakout.py]: Yes, in this PR.

(Although the softmax action selection works, I don't think the breakout scripts actually learn anything. I didn't run enough episodes to test this.)

File: examples/breakout/breakout_stdp.py (hunk 2 above)
Comment (sharath) [reply examples/breakout/breakout_stdp.py]: Yes, in this PR.

File: bindsnet/pipeline/environment_pipeline.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ELanning) [reply bindsnet/pipeline/environment_pipeline.py]: https://github.com/Hananel-Hazan/bindsnet/blob/626ae8397063fae542917abfc223b0491d46fcba/bindsnet/network/nodes.py#L23
https://github.com/Hananel-Hazan/bindsnet/blob/8e1677bb3b5b45b793572db0c68d332b64acb0a6/examples/mnist/conv_mnist.py#L17

In other places it seems to be `input`.
